dali/operators/generic/constant.h

```diff
@@ -0,0 +1,107 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_CONSTANT_H_
+#define DALI_OPERATORS_GENERIC_CONSTANT_H_
+
+#include <vector>
+
+#include "dali/pipeline/operator/operator.h"
+#include "dali/core/tensor_view.h"
+#include "dali/core/static_switch.h"
+
+#define CONSTANT_OP_SUPPORTED_TYPES \
+  (bool, int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float)
+
+namespace dali {
+
+template <typename Backend>
+class Constant : public Operator<Backend> {
+ public:
+  using Base = Operator<Backend>;
+  using Workspace = workspace_t<Backend>;
+
+  explicit Constant(const OpSpec &spec) : Operator<Backend>(spec) {
+    spec.TryGetRepeatedArgument<int>(shape_arg_, "shape");
+    output_type_ = spec.GetArgument<DALIDataType>("dtype");
+    if (spec.HasArgument("fdata")) {
+      DALI_ENFORCE(!spec.HasArgument("idata"), "Constant node: `fdata` and `idata` arguments are "
+        "mutually exclusive");
+      fdata_ = spec.GetRepeatedArgument<float>("fdata");
+      if (shape_arg_.empty()) {
+        shape_arg_.push_back(fdata_.size());
+      } else {
+        DALI_ENFORCE(fdata_.size() == static_cast<size_t>(volume(shape_arg_)) || fdata_.size() == 1,
+          "The number of values does not match the shape specified");
+      }
+
+      if (!spec.HasArgument("dtype"))
+        output_type_ = DALI_FLOAT;
+    } else {
+      DALI_ENFORCE(spec.HasArgument("idata"),
+          "Constant node: either `fdata` or `idata` must be present.");
+
+      if (!spec.HasArgument("dtype"))
+        output_type_ = DALI_INT32;
+
+      idata_ = spec.GetRepeatedArgument<int>("idata");
+      if (shape_arg_.empty()) {
+        shape_arg_.push_back(idata_.size());
+      } else {
+        DALI_ENFORCE(idata_.size() == static_cast<size_t>(volume(shape_arg_)) || idata_.size() == 1,
+          "The number of values does not match the shape specified");
+      }
+    }
+    layout_ = spec.GetArgument<TensorLayout>("layout");
+    if (!layout_.empty()) {
+      DALI_ENFORCE(layout_.size() == static_cast<int>(shape_arg_.size()), make_string(
+        "Constant node: The requested layout \"", layout_, "\" has dimensionalilty which is "
+        "incompatible with the requested output shape."));
+    }
+  }
+
+  bool CanInferOutputs() const override {
+    // Return false, because we specifically don't want the executor to allocate
+    // the storage for the output - even though we can infer the shape.
+    return false;
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
+    output_desc.resize(1);
+    if (output_shape_.empty()) {
+      int batch_size = this->spec_.template GetArgument<int>("batch_size");
+      output_shape_ = uniform_list_shape(batch_size, shape_arg_);
+    }
+    output_desc[0] = { output_shape_, TypeTable::GetTypeInfo(output_type_) };
```

Comment:
If we do not want executor to allocate for us, why do we bother with this function?

Reply:
True.